samcli/commands/deploy/command.py

```diff
@@ -107,6 +108,23 @@
     required=False,
     is_flag=True,
     help="Preserves the state of previously provisioned resources when an operation fails.",
+    cls=ClickMutex,
+    incompatible_params=["on_failure"],
+)
+@click.option(
+    "--on-failure",
+    default="ROLLBACK",
+    type=click.Choice(["ROLLBACK", "DELETE", "DO_NOTHING"]),
+    required=False,
+    help="""
+    Provide an action to determine what will happen when a stack fails to create. Three actions are available:\n
+    - ROLLBACK: This will rollback a stack to a previous known good state.\n
+    - DELETE: The stack will rollback to a previous state if one exists, otherwise the stack will be deleted.\n
+    - DO_NOTHING: The stack will not rollback or delete.\n
```

Comment:
Can we mention the incompatibility with disable-rollback flag in the help text? Also can we have a comment here explaining that DO_NOTHING is exactly the same thing as disable rollback? Quoting from CFN documentation:

> To preserve successfully provisioned resources during a stack create operation
Specify the disable-rollback option or on-failure DO_NOTHING enumeration during a [create-stack](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/create-stack.html) operation.